### ISSUE TITLE
Homebridge crash on unauthorized connection attempt

### ISSUE THREAD

**Issue by oldschool777** [gbro115/homebridge-roomba690#3]
I have my 690 in my basement, and from time-to-time it drops the connection to wifi, or is slow to respond.  In the iRobot Home app, it shows as connected through the cloud, when it can't connect locally.  The problem, is that when it is in this state, and not available on the local network, the roomba960 plugin doesn't seem to handle the connection refusal gracefully, and crashes the whole Homebridge instance.  Here is the Homebridge error log of the event:

Thu, 28 Dec 2017 01:13:13 GMT Accessory [Homebridge] Getting value for Characteristic "On"
Thu, 28 Dec 2017 01:13:13 GMT Accessory [Homebridge] Getting value for Characteristic "Status Low Battery"
Thu, 28 Dec 2017 01:13:13 GMT Accessory [Homebridge] Got Characteristic "Status Low Battery" value: 0
/usr/local/lib/node_modules/homebridge-roomba690/node_modules/dorita980/lib/v2/local.js:30
    throw e;
    ^

Error: Connection refused: Not authorized
    at MqttClient._handleConnack (/usr/local/lib/node_modules/homebridge-roomba690/node_modules/mqtt/lib/client.js:871:15)
    at MqttClient._handlePacket (/usr/local/lib/node_modules/homebridge-roomba690/node_modules/mqtt/lib/client.js:319:12)
    at work (/usr/local/lib/node_modules/homebridge-roomba690/node_modules/mqtt/lib/client.js:261:12)
    at Writable.writable._write (/usr/local/lib/node_modules/homebridge-roomba690/node_modules/mqtt/lib/client.js:271:5)
    at doWrite (/usr/local/lib/node_modules/homebridge-roomba690/node_modules/readable-stream/lib/_stream_writable.js:406:64)
    at writeOrBuffer (/usr/local/lib/node_modules/homebridge-roomba690/node_modules/readable-stream/lib/_stream_writable.js:395:5)
    at Writable.write (/usr/local/lib/node_modules/homebridge-roomba690/node_modules/readable-stream/lib/_stream_writable.js:322:11)
    at TLSSocket.ondata (_stream_readable.js:642:20)
    at TLSSocket.emit (events.js:159:13)
    at addChunk (_stream_readable.js:265:12)
    at readableAddChunk (_stream_readable.js:252:11)
    at TLSSocket.Readable.push (_stream_readable.js:209:10)
    at TLSWrap.onread (net.js:598:20)

Please let me know if there is any other information that I can collect to aid in fixing this issue.  I'm going to have to remove the 690 from my Homebridge for now, so it does not block my other devices from operating correctly.

**Comment by oldschool777** [comment]
Update:  After a bit more troubleshooting, I realized that the crash would only happen when I requested status in some way -- either via the Home app, or through Siri.  I then looked back at the error, and realized it was a "Not authorized" error, versus a failure to connect.  I had to reconfigure the wifi connection on the Roomba, so I'm guessing that it somehow lost the password used by this plugin, and that is the cause for the "Not authorized".  Once I went through the process of recreating a new password, updating my configuration, and restarting Homebridge, everything seems to be working correctly.

So, the good news is that the issue should not be as widespread, however, more error trapping around that connection process might be good to handle unexpected errors more gracefully.

Thanks for the great plugin!  This thing is awesome!  I wish the other Roomba plugin had the battery support.  I'm using it for the 960 that I just got, and this one for the 690 I got as well.  Having the battery status is really nice!
